File: Builder/Builder.java
class Pizza {
    private final int type;
    private final String toppings;
    private final String seasoning;

    public String toString() {
        return this.type + " " + this.toppings + " " + this.seasoning;
    }

    private Pizza(Builder builder) {
        this.type = builder.type;
        this.toppings = builder.toppings;
        this.seasoning = builder.seasoning;
    }

    public static class Builder {
        private final int type;
        private String toppings;
        private String seasoning;

        public Builder(int type) {
            this.type = type;
        }

        public Builder toppings(String t) {
            this.toppings = t;
            return this;
        }

        public Builder seasoning(String s) {
            this.seasoning = s;
            return this;
        }

        public Pizza build() {
            return new Pizza(this);
        }
    }
}

File: Builder/Builder.ts
class Pizza {
    
    private type: number;
    private toppings: String;
    private seasoning: String;

    private constructor(builder) {
        this.type = builder.type;
        this.toppings = builder.toppings;
        this.seasoning = builder.seasoning;
    }

    static get Builder() {
        class Builder {
            
            private type: number;
            private seasoning: String = "";
            private toppings: String = "";

            constructor(type: number) {
                this.type = type;
            }

            setSeasoning(s: String) {
                this.seasoning = s;
                return this;
            }

            setToppings(s: String) {
                this.toppings = s;
                return this;
            }

            build() {
                return new Pizza(this);
            }
        }
        return Builder;
    }
}

const pizza = new Pizza.Builder(4).setToppings("Ham").build();
console.log(pizza)